src/main/java/string_add_calculator/model/Numbers.java

```diff
@@ -15,16 +15,20 @@ private Numbers(final List<Number> numbers) {
 
     public static Numbers create(final String[] numbersString) {
         List<Number> numbers = Arrays.stream(numbersString)
-                .map(Number::newInstance)
+                .map(Number::of)
                 .collect(Collectors.toList());
 
         return new Numbers(numbers);
     }
 
-    public int sum() {
-        return numbers.stream()
-                .mapToInt(Number::toInt)
-                .sum();
+    public Number sum() {
+        Number result = Number.of(0);
```

Comment:
똑같은 기능의 객체를 매번 생성하기보다는 객체 하나를 재사용하는 것은 어떨까요?

```suggestion
        Number result = Number.ZERO;
```